rust/lance/src/dataset/write/merge_insert.rs

```diff
@@ -374,10 +374,29 @@ impl MergeInsertBuilder {
                 location!(),
             ));
         }
+
+        // Resolve column names using case-insensitive matching to handle
```

Comment:
This doesn't seem to have consistent "try exact first then fallback to inexact" semantics?